### PR TITLE
ign -> gz Namespace migration (narrow) : gz-common

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Part of https://github.com/gazebo-tooling/release-tools/issues/711, more narrow alternative to #356

## Summary

This is an alternative to #356 (actually using some commits from that branch) that is narrower in scope. It avoids most macro changes in an attempt to simplify the review process.

Once these namespace changes start being merged, things will break, so we will try to merge them all relatively close together in time.

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by methylDragon** [comment]
I think this one needs the commits adding the `#include <.../config.hh>`  commits

https://github.com/gazebosim/gz-common/pull/356/commits/9b074f87d7cd95be28e2369685c03f9798e08d25
https://github.com/gazebosim/gz-common/pull/356/commits/5aabd9089f088bd16d15d20d17741a032e9e65ee

**Comment by scpeters** [comment]
> I think this one needs the commits adding the `#include <.../config.hh>` commits
> 
> [9b074f8](https://github.com/gazebosim/gz-common/commit/9b074f87d7cd95be28e2369685c03f9798e08d25) [5aabd90](https://github.com/gazebosim/gz-common/commit/5aabd9089f088bd16d15d20d17741a032e9e65ee)

thanks! I just cherry-picked them

File: test/integration/deprecated_TEST.cc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+IGN_UTILS_WARN_IGNORE__DEPRECATED_DECLARATION
+
+/////////////////////////////////////////////////
+// Make sure the ignition namespace still works
+TEST(Deprecated, IgnitionNamespace)
+{
+  ignition::common::Battery battery;
+}
+
+IGN_UTILS_WARN_RESUME__DEPRECATED_DECLARATION
```
Comment (methylDragon) [inline test/integration/deprecated_TEST.cc:32]:
```suggestion
#include <gtest/gtest.h>

#define SUPPRESS_IGNITION_HEADER_DEPRECATION

#include <ignition/common/Battery.hh>
#include <ignition/utils/SuppressWarning.hh>

/////////////////////////////////////////////////
// Make sure the ignition namespace still works
TEST(Deprecated, IgnitionNamespace)
{
  ignition::common::Battery battery;
}

#undef SUPPRESS_IGNITION_HEADER_DEPRECATION
```

**Review by methylDragon — CHANGES_REQUESTED** [review]
This was from my dirty commits making it hard to isolate the deprecated_TEST change :bow:

**Comment by chapulina** [comment]
#356 got in

**Pull request closed without merging**